packages/core/src/lib/waku_peer_exchange/peer_discovery.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  PeerDiscovery,
+  PeerDiscoveryEvents,
+  symbol,
+} from "@libp2p/interface-peer-discovery";
+import { PeerInfo } from "@libp2p/interface-peer-info";
+import { EventEmitter } from "@libp2p/interfaces/dist/src/events";
+import debug from "debug";
+
+import { multiaddrsToPeerInfo } from "../multiaddr_to_peer_info";
+
+import { WakuPeerExchange } from ".";
+
+const log = debug("waku:peer-discovery-dns");
+
+export class PeerExchangeDiscovery
+  extends EventEmitter<PeerDiscoveryEvents>
+  implements PeerDiscovery
+{
+  private _started: boolean;
+
+  constructor(public peerExchange: WakuPeerExchange, public numPeers: number) {
+    super();
+    this._started = false;
+  }
+
+  /**
+   * Start discovery process
+   */
+  async start(): Promise<void> {
+    log("Starting peer discovery via Peer Exchange");
```

Comment:
```suggestion
    log("Starting peer discovery via Waku Peer Exchange");
```

there is a "Relay Peer Exchange" too in libp2p so best to qualify.